### PR TITLE
Fix #306512: Improvements to the splash screen

### PR DESCRIPTION
**Pull request by Spire42**
Resolves: [#306512](https://musescore.org/en/node/306512)

Rewrote the splash-screen code so that it renders everything on the fly using vector graphics instead of using pre-rendered images.

Also designed and implemented a darker version of the splash screen that is automatically used for unstable/development builds.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by mattmcclinch** [comment]
If you rename `Maintenance.svg` to `maintenance.svg`, this might actually build on filesystems where case matters.

**Comment by Spire42** [comment]
Thanks; fixed.

**Comment by Jojo-Schmitz** [comment]
Now Travis CI got lost once again

File: mscore/mssplashscreen.cpp [hunk 1]
```diff
@@ -19,19 +19,235 @@
...
+      _devBuildTextRect(scaleRect(designDevBuildTextRect)),
+      _logotypeRect(scaleSvgRect(designLogotypeRect, _logotypeRenderer)),
+      _messageTextRect(scaleRect(designMessageTextRect)),
+      _miscTextRect(scaleRect(designMiscTextRect))
+      {
+      setWindowTitle("MuseScore Startup");
+
+#ifdef Q_OS_MAC
+      // To have session dialog on top of splash screen on Mac.
+      sc->setWindowFlags(Qt::FramelessWindowHint);
```
Comment (mattmcclinch) [inline mscore/mssplashscreen.cpp:66]: Remove the `sc->`.

Comment (Spire42) [reply mscore/mssplashscreen.cpp]: D'oh! Fixed. Thanks.

File: mscore/mssplashscreen.cpp [hunk 2]
```diff
@@ -19,19 +19,235 @@
...
+const std::tuple<QColor, QColor> MsSplashScreen::stableBuildGradientColors   { "#0885DD", "#0C70B6" };
+const std::tuple<QColor, QColor> MsSplashScreen::unstableBuildGradientColors { "#04426E", "#063759" };
+
+const QColor MsSplashScreen::textColor { "#CCFFFFFF" };
+
+//---------------------------------------------------------
+//   MsSplashScreen constructor
+//---------------------------------------------------------
+
+MsSplashScreen::MsSplashScreen() :
```
Comment (Howard-C) [inline mscore/mssplashscreen.cpp:49]: Please put `:` in the next line.

Comment (Spire42) [reply mscore/mssplashscreen.cpp]: Done.

**Comment by Marr11317** [comment]
Would someone mind posting a screenshot? :)

**Comment by mattmcclinch** [comment]
Stable version:
stable
Unstable version:
unstable

**Pull request merged**